### ISSUE TITLE
Question: Is this suitable for Xamarin integration?

### ISSUE THREAD

**Issue by IainS1986** [segmentio/Analytics.NET#173]
Hi

I'm experimenting with setting up Segment integration with a Xamarin Native project.

I know theres a Segment Xamarin project here (https://github.com/segmentio/Analytics.Xamarin) but this appears to be all but abandoned, isn't in nuget or in the correct .netstandard setup anyway.

Instead I grabbed the nuget for this and have integrated it that way and things 'appear' to be working, but with maybe some gaps that I'm not sure. 

The original Xamarin integration was ServerSide anyway so I'm assuming using this nuget instead is fine? I found that IP address was not being set in Segment until I 'faked' the library version in the context to be "Analytics.Xamarin", then IP address was being set by the segment server correctly.

I'm also having to manually build all the information I can think of in the Segment Context (Device information, OS information, Screen information etc), but looking at the Xamarin project, it doesn't seem to do *anything* specific for Android or iOS in terms of filling out that information anyway?

Page and Screen views don't seem to go through to Google Analytics (UA) at all, wondering if thats again an issue with me using this nuget over the old Xamarin project, but again - the code in the old Xamarin project doesn't seem to do anything special in those areas?

Any advise would be greatly appreciated!

**TLDR; The old Xamarin project seems abandoned, is it fine to just use this and are there any further steps I should jump through to get it fully working (like with me faking the context.library.name)**

**Comment by IainS1986** [comment]
As an example if it helps, this is the sort of Raw event I'm generating in Xamarin with this Nuget (and my own context generating)...seems good to me but it would be helpful to know if this is viable (and why maybe GA is not tracking any of my Page or Screen calls)?

Should I change the Source in Segment to not be Xamarin?

{
  "anonymousId": null,
  "context": {
    "app": {
      "build": 0,
      "name": "XXXX",
      "namespace": "com.XXXX.XXXX",
      "version": "1.0 - DEBUG"
    },
    "device": {
      "adTrackingEnabled": false,
      "manufacturer": "Apple",
      "model": "iPhone8,1",
      "name": "XXXX’s iPhone",
      "type": "ios"
    },
    "integrations": {
      "All": "True",
      "Salesforce": "False"
    },
    "ip": "XXX.XXX.XXX.XXX",
    "library": {
      "name": "Analytics.Xamarin",
      "version": "3.7.1"
    },
    "locale": "en_GB",
    "os": {
      "name": "iOS",
      "version": "14.4.2"
    },
    "screen": {
      "density": 2,
      "height": 1334,
      "width": 750
    },
    "userAgent": "Mozilla/5.0 (iPhone; CPU iPhone OS 14_4_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148"
  },
  "event": "XXXX",
  "integrations": {},
  "messageId": "XXXXXX-f186-4545-b43a-XXXXXXX",
  "originalTimestamp": "2021-05-07T11:11:07.7269180+01:00",
  "properties": {
    "Level": "XXXX",
    "Occurrences": "1",
    "Type": "XXXX"
  },
  "receivedAt": "2021-05-07T10:11:16.006Z",
  "sentAt": "2021-05-07T10:11:15.882Z",
  "timestamp": "2021-05-07T10:11:07.849Z",
  "type": "track",
  "userId": "60275",
  "writeKey": "XXXXXXXX"
}

**Comment by juliette-ruchel** [comment]
Hi @IainS1986,

The Analytics.NET library should work fine with your Xamarin project as it is compatible with .NET Standard 1.3 and 2.0. This allows the library to be added to a Xamarin project without any problem. If your application is new then we recommend you to use this library instead of the Analytics.Xamarin one. That library is left only to support applications which already have it integrated.

Regarding your issue with Google Analytics, check on https://segment.com/docs/connections/sources/catalog/libraries/server/net/ to see if it helps you configure the Options property with the SetIntegration only to Google Analytics perhaps. Also rembember to configure your Google Analytics UA property in the Destinations tab in your Segment Workspace. There you should Add a new Destination and Select Google Analytics. Once you finished following the steps, remember to add in the Serverside Tracking ID your google Analytics Property Tracking ID (e.g: UA-XXXXX-X). 

Hope you find this information helpful, let me know if you still have any trouble configuring it.

**Comment by nd4p90x** [comment]
@IainS1986 if you need to re-open this issue, please feel free to do so.  Thank you
